src/components/ProjectListItem/ProjectListItem.tsx

```diff
@@ -0,0 +1,209 @@
+    project,
+    onTagClick,
+    onClickProvider,
+    selectedResolver,
+    onCollapsedChange,
+    onGoalsCollapsedChange,
+    children,
+    loading = false,
+    goals,
+    deep = 0,
+}) => {
+    const [collapsed, setIsCollapsed] = useState(true);
+    const [collapsedGoals, setIsCollapsedGoals] = useState(true);
+
+    const offset = collapseOffset * (collapsed ? deep - 1 : deep);
+
+    const childs = useMemo(() => project.children.map(({ id }) => id), [project]);
+    const content = collapsed ? null : children?.(childs, deep + 1);
+
+    const onClickEnabled = children && childs.length;
+
+    const onClick = useCallback(() => {
+        if (onClickEnabled) {
+            setIsCollapsed((value) => !value);
+        }
+    }, [onClickEnabled]);
+
+    useEffect(() => {
+        onCollapsedChange?.(collapsed);
+    }, [collapsed, onCollapsedChange]);
+
+    useEffect(() => {
+        onGoalsCollapsedChange?.(collapsedGoals);
+    }, [collapsedGoals, onGoalsCollapsedChange]);
+
+    const onHeaderButtonClick = useCallback((e: MouseEvent) => {
+        e.stopPropagation();
+
+        setIsCollapsedGoals((value) => !value);
+    }, []);
+
+    return (
+        <Collapsable
+            collapsed={collapsed || loading}
+            onClick={onClick}
+            header={
+                <ProjectListContainer offset={offset}>
+                    <ProjectListItem
+                        title={project.title}
+                        owner={project.activity}
+                        participants={project.participants}
+                        starred={project._isStarred}
+                        watching={project._isWatching}
+                    >
+                        <StyledGoalsButton onClick={onHeaderButtonClick} text={tr('Goals')} />
+                    </ProjectListItem>
+                </ProjectListContainer>
+            }
+            content={content}
+            deep={deep}
+        >
+            {!collapsedGoals && (
+                <GoalsListContainer offset={offset}>
+                    {nullable(goals, (goals) =>
+                        goals.map((g) => (
+                            <GoalListItem
+                                createdAt={g.createdAt}
+                                updatedAt={g.updatedAt}
+                                id={g.id}
+                                shortId={g._shortId}
+                                projectId={g.projectId}
+                                state={g.state!}
+                                title={g.title}
+                                issuer={g.activity!}
+                                owner={g.owner!}
+                                tags={g.tags}
+                                priority={g.priority!}
+                                comments={g._count?.comments}
+                                estimate={g.estimate?.length ? g.estimate[g.estimate.length - 1] : undefined}
```

Comment:
`g._lastEstimate`

Reply:
we also probably should be able to use [Array#at](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at)